client/CServerHandler.cpp

```diff
@@ -573,18 +573,36 @@ void CServerHandler::sendRestartGame() const
 	sendLobbyPack(endGame);
 }
 
-void CServerHandler::sendStartGame(bool allowOnlyAI) const
+bool CServerHandler::validateGameStart(bool allowOnlyAI) const
 {
 	try
 	{
 		verifyStateBeforeStart(allowOnlyAI ? true : settings["session"]["onlyai"].Bool());
 	}
-	catch (const std::exception & e)
+	catch(CModHandler::Incompatibility & e)
+	{
+		logGlobal->warn("Incompatibility exception during start scenario: %s", e.what());
+
+		auto errorMsg = CGI->generaltexth->translate("vcmi.server.errors.modsIncompatibility") + '\n';
+		errorMsg += e.what();
+
+		showServerError(errorMsg);
+		return false;
+	}
+	catch(std::exception & e)
 	{
+		logGlobal->error("Exception during startScenario: %s", e.what());
 		showServerError( std::string("Unable to start map! Reason: ") + e.what());
```

Comment:
What about specific errors ExceptionNoTemplate and ExceptionNoHuman?
For them there are error messages from original game: CGI->generaltexth->allTexts[530] and CGI->generaltexth->allTexts[751]